diff --git a/src/xfpm-button.c b/src/xfpm-button.c
index ed72ea8..6dec7a7 100644
--- a/src/xfpm-button.c
+++ b/src/xfpm-button.c
@@ -215,7 +215,7 @@ xfpm_button_setup (XfpmButton *button)
     if (xfpm_button_xevent_key (button, XF86XK_MonBrightnessDown, BUTTON_MON_BRIGHTNESS_DOWN) )
 	button->priv->mapped_buttons |= BRIGHTNESS_KEY_DOWN;
 
-    if (xfpm_button_xevent_key (button, XF86XK_Battery, BUTTON_BATTERY))
+    if (xfpm_button_xevent_key (button, XF86XK_ScreenSaver, BUTTON_BATTERY))
         button->priv->mapped_buttons |= BATTERY_KEY;
 
     if ( xfpm_button_xevent_key (button, XF86XK_KbdBrightnessUp, BUTTON_KBD_BRIGHTNESS_UP) )
